src/nupic/utils/MovingAverage.cpp

```diff
@@ -32,53 +32,53 @@ using namespace::nupic;
 using namespace nupic::util;
 
 
-MovingAverage::MovingAverage(UInt wSize, const vector<Real32>& historicalValues)
-    : windowSize_(wSize)
+MovingAverage::MovingAverage(UInt wSize, const vector<Real>& historicalValues)
+    : slidingWindow_(wSize, historicalValues)
 {
-  if (historicalValues.size() != 0)
-  {
-    copy(
-      historicalValues.begin() + historicalValues.size() - wSize,
-      historicalValues.end(),
-      back_inserter(slidingWindow_));
-  }
-  total_ = Real32(accumulate(slidingWindow_.begin(), slidingWindow_.end(), 0));
+ auto window = slidingWindow_.getData();
+  total_ = Real(accumulate(begin(window), end(window), 0));
 }
 
 
-MovingAverage::MovingAverage(UInt wSize) : windowSize_(wSize), total_(0) {}
+MovingAverage::MovingAverage(UInt wSize) : 
+  slidingWindow_(wSize), total_(0) {}
+
+
+MovingAverage::MovingAverage(const SlidingWindow<Real>& internalSlidingWindow) :
+  slidingWindow_(internalSlidingWindow) {
+ auto window = slidingWindow_.getData();
```

Comment:
Similar to above, explicitly write `const std::vector<Real>&` rather than `auto` so that it's clear that this is accessing the vector by reference.